tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2064,6 +2169,309 @@ int BPF_KPROBE(trace_cap_capable)
     return 0;
 };
 
+SEC("kprobe/security_socket_create")
+int BPF_KPROBE(trace_security_socket_create)
+{
+    // trace the event security_socket_create
+
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SECURITY_SOCKET_CREATE, 4 /*argnum*/, 0 /*ret*/);
+
+    // getting event tags
+    u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
+    if (!tags) {
+        return -1;
+    }
+
+    int family = (int)PT_REGS_PARM1(ctx);
+    int type = (int)PT_REGS_PARM2(ctx);
+    int protocol = (int)PT_REGS_PARM3(ctx);
+    int kern = (int)PT_REGS_PARM4(ctx);
+
+    save_to_submit_buf(submit_p, (void *)&family, sizeof(int), INT_T, DEC_ARG(0, *tags));
+    save_to_submit_buf(submit_p, (void *)&type, sizeof(int), INT_T, DEC_ARG(1, *tags));
+    save_to_submit_buf(submit_p, (void *)&protocol, sizeof(int), INT_T, DEC_ARG(2, *tags));
+    save_to_submit_buf(submit_p, (void *)&kern, sizeof(int), INT_T, DEC_ARG(3, *tags));
+
+    events_perf_submit(ctx);
+    return 0;
+};
+
+SEC("kprobe/security_socket_listen")
+int BPF_KPROBE(trace_security_socket_listen)
+{
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    // getting source details
+
+    u16 family;
+    struct sock *sk;
+
+    struct socket *sock = (struct socket *)PT_REGS_PARM1(ctx);
+    int backlog = (int)PT_REGS_PARM2(ctx);
+
+    // getting struct sock from socket
+    bpf_probe_read(&sk, sizeof(sk), &sock->sk);
+
+    // getting socket family
+    bpf_probe_read(&family, sizeof(family), &sk->sk_family);
+    if ( (family != AF_INET) && (family != AF_INET6) ) {
+        return 0;
+    }
+
+    context_t context = init_and_save_context(ctx, submit_p, SECURITY_SOCKET_LISTEN, 2 /*argnum*/, 0 /*ret*/);
+
+    // getting event tags
+    u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
+    if (!tags) {
+        return -1;
+    }
+
+    if ( family == AF_INET ){
+
+        net_conn_v4_t net_details = {};
+
+        get_network_details_from_sock_v4(sk, &net_details, 0);
+
+        struct sockaddr_in local;
+        local.sin_family = family;
+        local.sin_port = net_details.local_port;
+        struct in_addr local_addr;
+        local_addr.s_addr = net_details.local_address;
+        local.sin_addr = local_addr;
```

Comment:
Why do we need local_addr var?
We can use instead:
local.sin_addr = net_details.local_address;

Reply:
we want the output to be of type `struct sockaddr_in`, and `sockaddr_in.sin_addr` is of type `struct in_addr`.
when we want to get the actual address we use: `sockaddr_in.sin_addr.s_addr`.

Reply:
So can't we just use `local.sin_addr.s_addr = net_details.local_address;` ?

Reply:
you're right